packages/core/src/test/notifications/rendering.test.ts

```diff
@@ -0,0 +1,235 @@
+    })
+
+    it('executes openTxt type button', async function () {
+        const testWindow = getTestWindow()
+        testWindow.onDidShowMessage((message) => {
+            // Simulate user clicking open txt type
+            message.selectItem('Read More')
+        })
+        const notification = getModalNotification()
+        await verifyTxtNotification(notification)
+    })
+})
+
+// generate test notifications
+function getToastURLTestNotification(): ToolkitNotification {
+    return {
+        id: 'test notification',
+        displayIf: {
+            extensionId: 'aws.toolkit.fake.extension',
+        },
+        uiRenderInstructions: {
+            content: {
+                [`en-US`]: {
+                    title: 'test',
+                    description: 'This is a url notification.',
+                    toastPreview: 'test toast preview',
+                },
+            },
+            onRecieve: 'toast',
+            onClick: {
+                type: 'openUrl',
+                url: 'https://aws.amazon.com/visualstudiocode/',
+            },
+        },
+    }
+}
+
+function getTxtNotification(): ToolkitNotification {
+    return {
+        id: 'test notification',
+        displayIf: {
+            extensionId: 'aws.toolkit.fake.extension',
+        },
+        uiRenderInstructions: {
+            content: {
+                [`en-US`]: {
+                    title: 'test',
+                    description: 'This is a text document notification.',
+                },
+            },
+            onRecieve: 'toast',
+            onClick: {
+                type: 'openTextDocument',
+            },
+        },
+    }
+}
+
+function getModalNotification(): ToolkitNotification {
+    return {
+        id: 'test notification',
+        displayIf: {
+            extensionId: 'aws.toolkit.fake.extension',
+        },
+        uiRenderInstructions: {
+            content: {
+                [`en-US`]: {
+                    title: 'test',
+                    description: 'This is a modal notification.',
+                },
+            },
+            onRecieve: 'modal',
+            onClick: {
+                type: 'modal',
+            },
+            actions: [
+                {
+                    type: 'updateAndReload',
+                    displayText: {
+                        'en-US': 'Update and Reload',
+                    },
+                },
+                {
+                    type: 'openUrl',
+                    url: 'https://aws.amazon.com/visualstudiocode/',
+                    displayText: {
+                        'en-US': 'Proceed to Wiki',
+                    },
+                },
+                {
+                    type: 'openTxt',
+                    displayText: {
+                        'en-US': 'Read More',
+                    },
+                },
+            ],
+        },
+    }
+}
+
+function getOpenExternalStub(): sinon.SinonStub {
```

Comment:
This is already stubbed in `globalSetup.test.ts`

Reply:
> that is why I created this function because we can't have two stubs with the same openExternal command
I don't follow.

> The other way is to export the stub from globalSetup.test.ts, is that more preferred?
Its already exported and is used by other tests